lib/puppet/util/filetype.rb

```diff
@@ -191,13 +191,21 @@ def path=(user)
 
     # Read a specific @path's cron tab.
     def read
+      unless Puppet::Util.uid(@path)
+        Puppet.debug _("The %{path} user does not exist. Treating their crontab file as empty in case Puppet creates them in the middle of the run.") % { path: @path }
+
+        return ""
+      end
+
       Puppet::Util::Execution.execute("#{cmdbase} -l", failonfail: true, combine: true)
     rescue => detail
       case detail.to_s
       when /no crontab for/
         return ""
       when /are not allowed to/
-        raise FileReadError, _("User %{path} not authorized to use cron") % { path: @path }, detail.backtrace
+        Puppet.err _("The %{path} user is not authorized to use cron. Their crontab file is treated as empty in case Puppet modifies e.g. cron.deny or cron.allow in the middle of the run.") % { path: @path }
```

Comment:
Any thoughts on the error message here? Was trying to keep it consistent with what the original crontab provider did.

Reply:
Could also make it a debug message to keep it consistent with the unknown user case.

Reply:
Talked to @branan -- this should be debug